src/recipe_dictionary.cpp

```diff
@@ -33,9 +33,57 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt ) cons
 {
     std::vector<const recipe *> res;
 
+    size_t split;
+    split = txt.find( ":" );
+    std::string key = "", value;
+    if( split != std::string::npos ) {
+        key = txt.substr( 0, split );
```

Comment:
This all looks a bit weird. Couldn't you just check `txt[1] == ":"` and then `switch( txt[0] )`? Obviously add some bounds checking.